stdr_robot/src/stdr_robot.cpp

```diff
@@ -442,25 +467,7 @@ namespace stdr_robot
         _previousPose.theta);
     odom.twist.twist = _motionControllerPtr->getVelocity();
 
-    _odomPublisher.publish(odom);
-
-    //!< Sensors tf
-    for (int i = 0; i < _sensors.size(); i++) {
-      geometry_msgs::Pose2D sensorPose = _sensors[i]->getSensorPose();
-
-      tf::Vector3 trans(sensorPose.x, sensorPose.y, 0);
-      tf::Quaternion rot;
-      rot.setRPY(0, 0, sensorPose.theta);
-
-      tf::Transform robotToSensor(rot, trans);
-
-      _tfBroadcaster.sendTransform(
-        tf::StampedTransform(
-          robotToSensor,
-          ros::Time::now(),
-          getName(),
-          _sensors[i]->getFrameId()));
-    }
+   _odomPublisher.publish(odom);
```

Comment:
In this function the same time can be used when publishing odometry and the pose tf, instead of `ros::Time::now()` twice, in order to avoid the issue that you mentioned in your ROS Answer.

Reply:
true. In fact, using the same timestamp also for the sensors made things better (though using static tf is much much better anyway and the reasonable thing to do)